lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -478,6 +509,32 @@ def add_customer_address(xml, options)
         end
       end
 
+      # Payment can be done through either Credit Card or Electronic Check
+      def add_payment_source(xml, payment_source, options = {})
+        if payment_source.instance_of?(ActiveMerchant::Billing::Check)
+          add_echeck(xml, payment_source, options)
+        else
+          add_creditcard(xml, payment_source, options[:currency])
+        end
+      end
+
+      # Adds Electronic Check attributes
+      def add_echeck(xml, check, options = {})
+        xml.tag! :CardBrand, 'EC'
+        xml.tag! :CurrencyCode, currency_code(options[:currency])
+        xml.tag! :CurrencyExponent, currency_exponents(options[:currency])
+        unless check.nil?
```

Comment:
I'm not sure what the general preference is here, but I'm of the opinion that we don't raise exceptions on data we send in the request. Given that AM is an abstraction for the gateways, we should pass data through as it is or isn't received and let the gateway dictate what is and isn't valid.

Reply:
Instead of creating an invalid request where the mandatory parameters are missing and then 
waiting for the server to send an expected error, simple validation has been added to save the unnecessary request
response roundtrip before the error is returned back to the caller. Moreover, validate function exists within active merchant
for multiple payment source objects.

Reply:
It's not necessarily a bad idea to validate data before passing it through, but for payment methods, it can be particularly challenging. Especially since we deal with payment methods from many different countries with their own set of financial standards. 

The `validate` method for check payment method type actually runs the `routing_number` against an algorithm that is specific to US banks. As far as I can tell, `validate` is not used in AM outside of test methods. It is probably better to maintain consistency throughout AM and follow the implementation for adding payment method types.

It's also important to remember that AM is an open source project and taking the agnostic approach gives us the flexibility of implementing a gateway and using it for our specific needs while also allowing other users of this project to build out their own use cases.

Reply:
Point taken. We will remove this validation.